debezium-core/src/main/java/io/debezium/transforms/debeziumtimestampcoverter/DebeziumTimestampConverter.java

```diff
@@ -0,0 +1,698 @@
+                result.setTimeInMillis(0L);
+                result.set(Calendar.HOUR_OF_DAY, origCalendar.get(Calendar.HOUR_OF_DAY));
+                result.set(Calendar.MINUTE, origCalendar.get(Calendar.MINUTE));
+                result.set(Calendar.SECOND, origCalendar.get(Calendar.SECOND));
+                result.set(Calendar.MILLISECOND, origCalendar.get(Calendar.MILLISECOND));
+                return result.getTime();
+            }
+
+            @Override
+            public Schema optionalSchema() {
+                return SchemaBuilder.int32()
+                        .name(org.apache.kafka.connect.data.Date.LOGICAL_NAME)
+                        .version(1).optional().build().schema();
+            }
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_INT64, () -> {
+            return Schema.OPTIONAL_INT64_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_INT32, () -> {
+            return Schema.OPTIONAL_INT32_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_INT16, () -> {
+            return Schema.OPTIONAL_INT16_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_INT8, () -> {
+            return Schema.OPTIONAL_INT8_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_FLOAT64, () -> {
+            return Schema.OPTIONAL_FLOAT64_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_FLOAT32, () -> {
+            return Schema.OPTIONAL_FLOAT32_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_BOOLEAN, () -> {
+            return Schema.OPTIONAL_BOOLEAN_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(STRING_TYPE, () -> {
+            return Schema.OPTIONAL_STRING_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_BYTES, () -> {
+            return Schema.OPTIONAL_BYTES_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_STRUCT, () -> {
+            return SchemaBuilder.type(Schema.Type.STRUCT).optional().build().schema();
+        });
+
+    }
+
+
+    public static Boolean containsField(String[] field_type, String field_name) {
+        return Stream.of(field_type).anyMatch(f -> f.equals(field_name));
+    }
+
+    // This is a bit unusual, but allows the transformation config to be passed to static anonymous classes to customize
+    // their behavior
+    public static class Config {
+        Config(String field, String type, SimpleDateFormat format, SimpleDateFormat dateFormat, String[] field_type, String[] struct_field) {
+            this.field = field;
+            this.type = type;
+            this.format = format;
+            this.dateFormat = dateFormat;
+            this.field_type = field_type;
+            this.struct_field = struct_field;
+        }
+
+        String field;
+        String type;
+        SimpleDateFormat format;
+        SimpleDateFormat dateFormat;
+        String[] field_type;
+        String[] struct_field;
```

Comment:
The fields should be defined before the constructor, and I'd suggest making them `public final`.